Sources/Client/ScoreboardView.cpp

```diff
@@ -49,17 +49,33 @@ namespace spades {
 		static const Vector4 spectatorTextColor = {220.f / 255, 220.f / 255, 0,
 		                                           1}; // Goldish yellow
 		static const auto spectatorTeamId = 255;       // Spectators have a team id of 255
+		
 
 		ScoreboardView::ScoreboardView(Client *client)
 		    : client(client), renderer(client->GetRenderer()) {
 			SPADES_MARK_FUNCTION();
-			ctf = NULL;
-			tc = NULL;
+			world = nullptr;
+			tc = nullptr;
+			ctf = nullptr;
+			image = nullptr;
+
+			// Use GUI font if spectator string has special chars
+			auto spectatorString = _TrN("Client", "Spectator{1}", "Spectators{1}", "", "");
+			auto has_special_char = 
+				std::find_if(spectatorString.begin(), spectatorString.end(),
+				[](char ch) {
+					ch = static_cast<unsigned char>(ch);
```

Comment:
Okay, I see, `isalnum` accepts a `int` and it must be representable as `unsigned char` (or is equal to `EOF`). But casting `ch` to `unsigned char` and then reassigning the result to `ch` doesn't make `ch` unsigned. In fact, this is what I get from LLDB:

```
(lldb) bt 1
* thread #1: tid = 0x1054e35, 0x00000001003733c2 OpenSpades`isalnum(_c=-24) + 18 at ctype.h:214, queue = 'com.apple.main-thread', stop reason = step in
    frame #0: 0x00000001003733c2 OpenSpades`isalnum(_c=-24) + 18 at ctype.h:214
```